BUILD.gn

```diff
@@ -204,6 +204,13 @@ if (current_toolchain != "${dir_pw_toolchain}/default:default") {
         }
         deps += [ "${chip_root}/src/controller/python:chip-repl" ]
       }
+      if (chip_can_build_all_clusters_app) {
+        if (is_libfuzzer) {
+          deps += [ "${chip_root}/examples/all-clusters-app/linux:chip-all-clusters-app-fuzzing" ]
```

Comment:
This seems unusual to me - could we instead build all-clusters as a separate build step? Why would chip_build_tools include a dependency into examples? Are there alternatives?

Reply:
See line 344 below on how enable_linux_* are used for unified build. I believe we probably want something similar rather than a tie to tools.gni

Reply:
Yes, we can decouple it from chip_build_tools.

If we build all-clusters-app separately, we will profile the common source files twice, one from unit test build and one from the all-clusters-app build,  maybe with different compile option.

We use .gcno file to track what source files are tracked. The .gcno notes file is generated when the source file is compiled with the GCC -ftest-coverage option. It contains information to reconstruct the basic block graphs and assign source line numbers to blocks, and this file is not merge-able.